### PR TITLE
Allow setting custom properties in service_bus send_message()

### PR DESCRIPTION
**Pull request by bryanpendleton**
Following on the recent broker_properties additions, it'd be nice to be able to set custom properties as well.

### CODE REVIEW

**Review by heaths — CHANGES_REQUESTED** [review]
I wanted to caution again about adding a lot of functionality to the code currently in `main`. It is unsupported, and we will be moving it to a new `legacy` branch soon. I recommend opening tracking issues to make sure you see features in future *supported* crates. I can't guarantee we will review all unsupported crates to migrate functionality to supported crates as we onboard services.

**Comment by bryanpendleton** [comment]
@microsoft-github-policy-service agree company="Fugu UX"

File: sdk/messaging_servicebus/src/service_bus/mod.rs [hunk 1]
```diff
@@ -315,6 +315,7 @@ impl FromStr for BrokerProperties {
 pub struct SendMessageOptions {
     pub content_type: Option<String>,
     pub broker_properties: Option<SettableBrokerProperties>,
+    pub custom_properties: Option<headers::Headers>,
```
[the remark this answers is not in the document]
Comment (bryanpendleton) [reply sdk/messaging_servicebus/src/service_bus/mod.rs]: Sure, replaced headers::Headers with just HashMap<String, String>.

**Comment by bryanpendleton** [comment]
> I wanted to caution again about adding a lot of functionality to the code currently in `main`. It is unsupported, and we will be moving it to a new `legacy` branch soon. I recommend opening tracking issues to make sure you see features in future _supported_ crates. I can't guarantee we will review all unsupported crates to migrate functionality to supported crates as we onboard services.

Thanks for the heads up! Is there a resource someplace documenting this plan, and how best to contribute? I don't know much about the history of these crates besides noting that they were largely originated from Microsoft but marked unofficial.

**Comment by heaths** [comment]
No, but we'll be making changes soon. If you want to see this change in that source, please open a separate tracking issue and reference this PR for context.

In the future, please push new commits so I can review the diff instead of reviewing the entire PR again.

**Review by heaths — APPROVED** [review]

**Pull request merged**